services/motion/builtin/builtin.go

```diff
@@ -316,3 +460,29 @@ func (ms *builtIn) GetPose(
 		supplementalTransforms,
 	)
 }
+
+// NewLocalizer constructs either a SLAMLocalizer or MovementSensorLocalizer from the given resource
+func (ms *builtIn) NewLocalizer(ctx context.Context, res resource.Resource) (localizer.Localizer, error) {
```

Comment:
My bad I led you astray with this one.  Since Limits are out of scope now there is not reason to have this be a part of a motion service and can instead live with the rest of the localizer code.  Then we can go ahead and make all the structs that implement the interface private

Reply:
I disagree because we would just be instantiating them through the new constructor method.  Part of the point of making this abstraction is that once we have a constructor we actually shouldn't care what type it is under the hood so we can treat them the same

Reply:
gotcha!